### PR TITLE
fix: show revision in badge when param is true

### PR DESCRIPTION
**Pull request by thijsvtol**
Signed-off-by: Thijs van Tol <43065692+thijsvtol@users.noreply.github.com>

Description:
If I use the api for retrieving the badges, the revision will always be shown when the `revision` query param is present.
So for example if you call `https://sample.com/api/badge?name=tomato-service?revision=false` it will return the badge WITH the revision.

So this PR fixes that it will only show the revision if the value is `true`

Checklist:

* [ ] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [x] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] Optional. My organization is added to USERS.md.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/master/community#contributing-to-argo)
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [x] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)).

### CODE REVIEW

File: server/badge/badge.go [hunk 1]
```diff
@@ -107,7 +107,7 @@ func (h *Handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		}
 	}
 	//Sample url: http://localhost:8080/api/badge?name=123&revision=true
-	if _, ok := r.URL.Query()["revision"]; ok && enabled {
+	if revisionParam, ok := r.URL.Query()["revision"]; ok && enabled && revisionParam[0] == "true" {
```
Comment (crenshaw-dev) [inline server/badge/badge.go:110]: Could we make the comparison case-insensitive? It's a breaking API change, but I think minor enough to be acceptable in a patch release. But I'd like to minimize potential breakage.

**Review by crenshaw-dev — CHANGES_REQUESTED** [review]

File: server/badge/badge.go (hunk 1 above)
Comment (thijsvtol) [reply server/badge/badge.go]: good one, applied it with `EqualFold`

**Comment by thijsvtol** [comment]
@crenshaw-dev please review and merge

**Review by crenshaw-dev — APPROVED** [review]
lgtm, thanks @thijsvtol!

Out of an abundance of caution, I'm only going to cherry-pick this into 2.5. If folks desperately need this in prior versions, I'll consult with #argo-contributors in Slack before cherry-picking.

**Comment by crenshaw-dev** [comment]
Cherry-picked onto release-2.5 for 2.5.0-rc2.

**Pull request merged**